### PR TITLE
Fix: report semantic error for STOP/exit code of wrong type (#12319)

### PR DESCRIPTION
**Pull request by usefahmed07**
**Fixes** #12319

### CODE REVIEW

File: tests/tests.toml [hunk 1]
```diff
@@ -505,6 +505,9 @@ asr = true
 [[test]]
 filename = "errors/real_unsupported_kind_01.f90"
 asr = true
+[[test]]
+filename = "errors/stop_code_wrong_type_01.f90"
+asr = true
```
Comment (certik) [inline tests/tests.toml]: Use continue compilation.

Comment (usefahmed07) [reply tests/tests.toml]: ok thanks

**Review by opixdown — APPROVED** [review]

**Comment by opixdown** [comment]
@usefahmed07 LGTM! just move the test to continue compilation. We generally add semantic error tests under continue compilation while keeping the diff minimum. Also, if you are using VS Code, you can use the source control tab to review exactly what youare committing before pushing. Thankyou

**Comment by opixdown** [comment]
remove blank line for every line you add thats the way to keep the minimal diff

File: tests/errors/continue_compilation_1.f90 [hunk 2]
```diff
@@ -657,6 +657,8 @@ program continue_compilation_1
     integer :: eoshift_bad_shift(2, 2)
     eoshift_bad_shift = 1
     b1 = eoshift(b1, eoshift_bad_shift)
+    stop_code_wrong_type_01_x = 5.0
```
Comment (opixdown) [inline tests/errors/continue_compilation_1.f90]: for this remove two lines if no space is there try to do it in subroutine and append it in eof

Comment (HarshitaKalani) [reply tests/errors/continue_compilation_1.f90]: Can you please address this @usefahmed07 ? This is to make the reference diff small so that it's easy to review and we make sure there're no unwanted changes.

**Review by HarshitaKalani — COMMENTED** [review]
Can you please address the comment and then mark it ready for review?

File: tests/errors/continue_compilation_1.f90 [hunk 3]
```diff
@@ -657,6 +657,8 @@ program continue_compilation_1
     integer :: eoshift_bad_shift(2, 2)
     eoshift_bad_shift = 1
     b1 = eoshift(b1, eoshift_bad_shift)
+
+
```
Comment (opixdown) [inline tests/errors/continue_compilation_1.f90]: remove this two line

Comment (usefahmed07) [reply tests/errors/continue_compilation_1.f90]: Done

File: src/lfortran/semantics/ast_body_visitor.cpp [hunk 4]
```diff
@@ -7697,6 +7706,21 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             }
         }
         if (!original_sym || (original_sym && is_external)) {
+            if (to_lower(sub_name) == "exit") {
+                diag.semantic_warning_label(
+                    "Routine `" + sub_name + "` is a non-standard function",
+                    {x.base.base.loc},
+                    ""
+                );
```
Comment (HarshitaKalani) [inline src/lfortran/semantics/ast_body_visitor.cpp:7736]: This error is not tested. Can you please add a test for this too?

**Review by HarshitaKalani — COMMENTED** [review]
There are two errors added in this PR but only one is tested. Can you please add a test for the other error as well? Thanks!

File: src/lfortran/semantics/ast_body_visitor.cpp (hunk 4 above)
Comment (usefahmed07) [reply src/lfortran/semantics/ast_body_visitor.cpp:7736]: ok no problem i am on it

**Comment by usefahmed07** [comment]
Hi @HarshitaKalani can you review it thanks!

**Comment by usefahmed07** [comment]
@assem2002
